### DISCUSSION TITLE
Searchable with join

### DISCUSSION THREAD

**Discussion by Epitaffio** [Power-Components/livewire-powergrid#1611] [Q&A]
### Have you searched through other issues to see if your problem is already reported or has been fixed?

Yes, I did not find it.

### Did you read the documentation?

Yes, I did not find it.

### Have you tried to publish the views?

No, this error is not related to views.

### Is there an error in the console?

No

### PHP Version

8.3

### PowerGrid

5.8.0

### Laravel

11.10

### Livewire

3.5.0

### Theme

Tailwind 3.x

### Describe the bug.

Hi, I am struggling to find why my implementation doesn't work.

For background II have 2 table that I have joined in the datasource (for the sake of testing I have added all the possible variations for my field):
```php 
    public function datasource(): Builder
    {
        return Cliente::query()
            ->select('clienti.*', 'agenti.AGDESAGE', 'agenti.AGDESAGE as agenti_desc', 'agenti.AGDESAGE as AGDESAGE')
            ->leftJoin('agenti', 'agenti.id',  'clienti.agente_id');
    }
```

In the fileds definition I have added all the variants:
```php 
    public function fields(): PowerGridFields
    {
        return PowerGrid::fields()
            ->add('agenti.AGDESAGE')
            ->add('AGDESAGE')
            ->add('agenti_desc');
    }
```

As  also in the columns method:
```php 
public function columns(): array
    {
        return [
            Column::make('Codice', 'ANCODICE')->sortable()->searchable(),

            Column::make('test0', 'agenti_desc')->searchable()->sortable(),
            Column::make('test1', 'agenti_desc', 'agenti.AGDESAGE')->searchable()->sortable(),
            Column::make('test2', 'agenti_desc', 'AGDESAGE')->searchable()->sortable(),
            Column::make('test3', 'agenti.AGDESAGE')->searchable()->sortable(),
            Column::make('test4', 'AGDESAGE')->searchable()->sortable(),
        ];
    }
```

The sql is generated correctly and I can order all the fields
```sql
select `clienti`.*, `agenti`.`AGDESAGE`, `agenti`.`AGDESAGE` as `agenti_desc`, `agenti`.`AGDESAGE` as `AGDESAGE` from `clienti` inner join `agenti` on `agenti`.`id` = `clienti`.`agente_id` order by `ANCODICE` asc limit 10 offset 0
```

but when I try to use the global search, the joined field are ignored:
```sql
select count(*) as aggregate from `clienti` inner join `agenti` on `agenti`.`id` = `clienti`.`agente_id` where ((`clienti`.`ANCODICE` LIKE '%aaaa%'))
```

I can't understand what I am doing different from the join example in the docs (https://demo.livewire-powergrid.com/examples/datasource-join).

Digging in the package code the only place where I can see maybe(?) an error is inside src/DataSource/Builder.php

on line 141
```php 
$columnList = $this->getColumnList($modelTable);
```

Where the method return the original table's field list but not the joined ones, so the consequential on line 152 gave false and the column is skipped
```php 
$hasColumn = isset($columnList[$field]);
```

But why in the join demo it works?

Hope someone can point me in the right direction, thanks!

**Comment by luanfreitasdev** [comment]
Unfortunately, I'm not sure what could be causing this behavior. For now, I'll move this to discussions

**Comment by vrerabek** [comment]
I am encountering the same problem where joined columns are not recognized when attempting to use ->searchable(). Unfortunately, I don't have a solution either, but I wanted to let you know that you're not the only one experiencing this.

- PHP Version 8.3
- PowerGrid Version 5.9.0
- Laravel Version 11.16
- Livewire Version 3.5.4
- Theme: Tailwind

**Comment by luanfreitasdev** [comment]
Hello

I think I figured out what's causing the problem. 
The problem is that in the demo, both the categories and dishes tables contain the same field name ("name"), so in your case, it doesn't fit the $hasColumn condition. I'll open a PR to fix this

```php
if ($hasColumn) {
    try {
        ds([$columnList, $field]);
        ds("{$table}.{$field}")->blue();
        
        $query->orWhere("{$table}.{$field}", Sql::like($query), "%{$search}%");
    } catch (\Throwable) {

        $query->orWhere("{$table}.{$field}", Sql::like($query), "%{$search}%");
    }
}
```
![image](https://github.com/user-attachments/assets/43ac6567-2df1-4e1b-b24f-b7b2bb7b9811)

Thanks
